app/test/search/text_utils_test.dart

```diff
@@ -105,7 +105,25 @@ Other useful methods will be added soon...
       });
     });
 
-    test('Cased words', () {
+    test('CamelCase extraction', () {
+      expect(extractCamelCase('CamelCaseWord').toList(),
+          ['Camel', 'Case', 'Word']);
+      expect(extractCamelCase('firstLowerCase').toList(),
+          ['first', 'Lower', 'Case']);
+      expect(extractCamelCase('snake_case_word').toList(), ['snake_case_word']);
+    });
+
+    test('isSelfDocumenting positive examples', () {
+      expect(isSelfDocumenting('todayBorderColor'), true);
+      expect(isSelfDocumenting('weekendTextStyle'), true);
```

Comment:
I would like these tests to have more examples to be able to see where the border goes between descriptive and nondescriptive names.

Reply:
I've added a few more test cases. The boundary also depends on whether the word was popular enough to get into the top 5000, or less so (e.g. if it is a technical term). I've considered to use a broader dictionary (100k+ English words), but I'm not sure if rare words would be as descriptive on their own as frequently used ones.

Reply:
My feeling is that they are less generic and therefore more descriptive.
theObject says almost nothing, while randomizedDictionary gives some clue of the purpose.

Reply:
I think what we don't want to encourage is to use the same words in the documentation that were already used in the API symbol. Such documentation feel trivial and unnecessary.

When an API symbol is already a commonly used word, chances are that its documentation won't be complex either. I understand that there will be false positives and false negatives in either case, and this can't be perfect, however, for the developers, this will boil down to:

1) Does pub nags me to do trivial things? (for authors)
2) Can I understand this package easily? (for library users)

This is more about decreasing the probability of (1), while we don't really compromise (2). I think the linked issue (#1881) clearly falls into the first category, I don't feel that most of the fields need additional documentation there.

Reply:
> When an API symbol is already a commonly used word, chances are that its documentation won't be complex either.

I suppose the counter argument is that the more common a word is the less information in conveys. For example `ImageWidth` of what image? in what unit?
Where as a less common word is more likely to be a domain specific term, which requires less explanation for people who understand it, and more explanantion for people who don't.

I'm not going to propose that we invert the heuristic... just slowly starting to wonder if looking at the frequency of the words is a good heuristic at all.. If it says anything useful.